### PR TITLE
Updating project / item template icons and descriptions

### PR DESCRIPTION
**Pull request by niels9001**
Closes: #4923

The following changes are included:

**Updates to the Windows App SDK and WinUI logo**
The default Windows icon that was used to represent WASDK is now replaced with a dedicated Windows App SDK logo. The old WinUI logo has been updated to the latest version.

**Updates to project and item template titles and descriptions**
- Replacing `Windows UI Library (WinUI 3)` with `WinUI 3`, as this is the product naming we are using across our docs and communication.
- Following the WPF / UWP template naming conventions starting with the framework name and then the project title. E.g. `Blank App, Packaged (WinUI 3 in Desktop)` => `WinUI Blank App (Packaged)`
- Changing `Custom Control` to `Templated Control`, which is the correct name for this item template.
- Making descriptions less verbose.

**Updates to project and item template icons**
Instead of using the default WinUI logo for every template, we are now using individual icons following the Visual Studio design guidelines for iconography (and similar to what WPF and UWP have been using). This helps the developer to understand what the template is about and what language it's using.

Mock-up: original vs. new
image

Mock-up: original vs. new
image

### CODE REVIEW

File: dev/VSIX/Extension/Cs/Common/VSPackage.resx [hunk 1]
```diff
@@ -151,60 +151,60 @@
...
-    <value>[Experimental] Class Library (WinUI 3 in Desktop)</value>
+    <value>[Experimental] Class Library</value>
   </data>
   <data name="1027" xml:space="preserve">
-    <value>A project for creating a managed class library (.dll) for Desktop apps based on the Windows UI Library (WinUI 3).</value>
+    <value>A project for creating a managed class library (.dll) for WinUI 3 apps.</value>
   </data>
   <data name="1029" xml:space="preserve">
-    <value>Blank App, Packaged with Windows Application Packaging Project (WinUI 3 in Desktop)</value>
+    <value>WinUI Blank App (Packaged with Windows Application Packaging Project)</value>
```
Comment (0x5bfa) [inline dev/VSIX/Extension/Cs/Common/VSPackage.resx:184]: What WAP stands for is described in its description.
```suggestion
    <value>WinUI Blank App (Packaged with WAP)</value>
```

File: dev/VSIX/Extension/Cs/Common/VSPackage.resx [hunk 2]
```diff
@@ -151,60 +151,60 @@
...
   </data>
   <data name="1030" xml:space="preserve">
-    <value>[Experimental] Blank App, Packaged with Windows Application Packaging Project (WinUI 3 in Desktop)</value>
+    <value>[Experimental] WinUI Blank App (Packaged with Windows Application Packaging Project)</value>
```
Comment (0x5bfa) [inline dev/VSIX/Extension/Cs/Common/VSPackage.resx:187]:
```suggestion
    <value>[Experimental] WinUI Blank App (Packaged with WAP)</value>
```

File: dev/VSIX/ProjectTemplates/Desktop/CSharp/PackagedApp/BlankApp/WinUI.Desktop.Cs.BlankApp.vstemplate [hunk 3]
```diff
@@ -2,8 +2,8 @@
 <VSTemplate Version="3.0.0" xmlns="http://schemas.microsoft.com/developer/vstemplate/2005" Type="Project">
   <TemplateData>
     <Hidden>true</Hidden>
-    <Name>Blank App (WinUI 3 in Desktop)</Name>
-    <Description>A project for creating a Desktop app based on the Windows UI Library (WinUI 3).</Description>
+    <Name>Blank App (WinUI)</Name>
```
Comment (0x5bfa) [inline dev/VSIX/ProjectTemplates/Desktop/CSharp/PackagedApp/BlankApp/WinUI.Desktop.Cs.BlankApp.vstemplate:5]:
```suggestion
    <Name>WinUI Blank App</Name>
```

File: dev/VSIX/ProjectTemplates/Desktop/CppWinRT/PackagedApp/BlankApp/WinUI.Desktop.CppWinRT.BlankApp.vstemplate [hunk 4]
```diff
@@ -2,8 +2,8 @@
 <VSTemplate Version="3.0.0" xmlns="http://schemas.microsoft.com/developer/vstemplate/2005" Type="Project">
   <TemplateData>
     <Hidden>true</Hidden>
-    <Name>Blank App (WinUI 3 in Desktop)</Name>
-    <Description>A project for creating a Desktop app based on the Windows UI Library (WinUI 3).</Description>
+    <Name>Blank App (WinUI)</Name>
```
Comment (0x5bfa) [inline dev/VSIX/ProjectTemplates/Desktop/CppWinRT/PackagedApp/BlankApp/WinUI.Desktop.CppWinRT.BlankApp.vstemplate:5]:
```suggestion
    <Name>WinUI Blank App</Name>
```

File: dev/VSIX/Extension/Cs/Common/VSPackage.resx [hunk 5]
```diff
@@ -118,31 +118,31 @@
     <value>System.Resources.ResXResourceWriter, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
   </resheader>
   <data name="1001" xml:space="preserve">
-    <value>Blank Window (WinUI 3 in Desktop)</value>
+    <value>Blank Window</value>
```
Comment (0x5bfa) [inline dev/VSIX/Extension/Cs/Common/VSPackage.resx:121]: Would it be better to append its extension like `Blank Window (.xaml/.cs)` or `Resource Dictionary (.xaml)`.

File: dev/VSIX/Extension/Cs/Common/VSPackage.resx (hunk 1 above)
Comment (Lamparter) [reply dev/VSIX/Extension/Cs/Common/VSPackage.resx:184]: Why not just
```suggestion
    <value>WinUI App (Packaged)</value>
```

File: dev/VSIX/ProjectTemplates/Desktop/CppWinRT/PackagedApp/BlankApp/WinUI.Desktop.CppWinRT.BlankApp.vstemplate (hunk 4 above)
Comment (Lamparter) [reply dev/VSIX/ProjectTemplates/Desktop/CppWinRT/PackagedApp/BlankApp/WinUI.Desktop.CppWinRT.BlankApp.vstemplate:5]:
```suggestion
    <Name>WinUI App</Name>
```

Comment (Lamparter) [reply dev/VSIX/ProjectTemplates/Desktop/CppWinRT/PackagedApp/BlankApp/WinUI.Desktop.CppWinRT.BlankApp.vstemplate:5]: Simpler = better

File: dev/VSIX/Extension/Cs/Common/VSPackage.resx (hunk 5 above)
Comment (Lamparter) [reply dev/VSIX/Extension/Cs/Common/VSPackage.resx:121]: Good idea - do other resources templates do this?

File: dev/VSIX/Extension/Cs/Common/VSPackage.resx (hunk 1 above)
Comment (0x5bfa) [reply dev/VSIX/Extension/Cs/Common/VSPackage.resx:184]: This is for single-packing project template, already exists.

Comment (Lamparter) [reply dev/VSIX/Extension/Cs/Common/VSPackage.resx:184]: Oops forgot it was the same

```suggestion
    <value>WinUI App (Packaged with Packaging Project)</value>
```

This sounds better imo

File: dev/VSIX/Extension/Cs/Common/VSPackage.resx (hunk 2 above)
Comment (Lamparter) [reply dev/VSIX/Extension/Cs/Common/VSPackage.resx:187]:
```suggestion
    <value>[Experimental] WinUI App (Packaged with Packaging Project)</value>
```

File: dev/VSIX/Extension/Cs/Common/VSPackage.resx (hunk 1 above)
Comment (0x5bfa) [reply dev/VSIX/Extension/Cs/Common/VSPackage.resx:184]: Well, no.

File: dev/VSIX/Extension/Cs/Common/VSPackage.resx (hunk 5 above)
Comment (niels9001) [reply dev/VSIX/Extension/Cs/Common/VSPackage.resx:121]: This is not common for item templates for other frameworks, and adds little value?

File: dev/VSIX/Extension/Cs/Common/VSPackage.resx (hunk 1 above)
Comment (niels9001) [reply dev/VSIX/Extension/Cs/Common/VSPackage.resx:184]: Agree that keeping Blank in the title makes sense; for alignment with other frameworks and for any potential other templates that might ship as part of WASDK.

@0x5bfa do you have examples where WAP is used vs. "Windows Application Packaging Project"?

Comment (Lamparter) [reply dev/VSIX/Extension/Cs/Common/VSPackage.resx:184]: It's really long though. 
Developers will understand the meaning of "WinUI App" and it's a simple name that represents WinUI.
Imagine if Class Library was called "Class Library as Runtime Assembly (Packaged as Link Library)" that would be too long.
I think it should be good to be simple

**Comment by Lamparter** [comment]
Something about "WinUI Blank App" sounds a bit off. "Blank WinUI App" is grammatically correct.

File: dev/VSIX/Extension/Cs/Common/VSPackage.resx (hunk 1 above)
Comment (0x5bfa) [reply dev/VSIX/Extension/Cs/Common/VSPackage.resx:184]: That was misleading, WAP isn't used across the platform (only Empty/Blank). But the folks have kept using that afaik.

For example: 
- https://developercommunity.visualstudio.com/t/windows-application-packaging-project-wap-causes-e/1347957
- Previous template
IMG_0705

Comment (0x5bfa) [reply dev/VSIX/Extension/Cs/Common/VSPackage.resx:184]: FYI, `WinUI Application` is consistent?

IMG_0706

Comment (niels9001) [reply dev/VSIX/Extension/Cs/Common/VSPackage.resx:184]: Ah, I see. Yeah things seem to be a bit different across WPF and UWP.

Does it make sense to keep things as close as possible to UWP?
image

File: dev/VSIX/Extension/Cpp/Common/VSPackage.resx [hunk 6]
```diff
@@ -118,31 +118,31 @@
...
-    <value>Blank Window (WinUI 3 in Desktop)</value>
+    <value>Blank Window</value>
   </data>
   <data name="1002" xml:space="preserve">
-    <value>[Experimental] Blank Window (WinUI 3 in Desktop)</value>
+    <value>[Experimental] Blank Window</value>
   </data>
   <data name="1003" xml:space="preserve">
-    <value>A blank window for Desktop apps based on the Windows UI Library (WinUI 3).</value>
+    <value>A blank window (WinUI 3).</value>
```
Comment (Lamparter) [inline dev/VSIX/Extension/Cpp/Common/VSPackage.resx:127]:
```suggestion
    <value>A blank window (WinUI).</value>
```

File: dev/VSIX/Extension/Cpp/Common/VSPackage.resx [hunk 7]
```diff
@@ -118,31 +118,31 @@
...
-    <value>Blank Page (WinUI 3)</value>
+    <value>Blank Page</value>
   </data>
   <data name="1006" xml:space="preserve">
-    <value>[Experimental] Blank Page (WinUI 3)</value>
+    <value>[Experimental] Blank Page</value>
   </data>
   <data name="1007" xml:space="preserve">
-    <value>A blank page for apps based on the Windows UI Library (WinUI 3).</value>
+    <value>A blank page (WinUI 3).</value>
```
Comment (Lamparter) [inline dev/VSIX/Extension/Cpp/Common/VSPackage.resx:136]:
```suggestion
    <value>A blank page (WinUI).</value>
```

File: dev/VSIX/Extension/Cpp/Common/VSPackage.resx [hunk 8]
```diff
@@ -118,31 +118,31 @@
...
-    <value>Resource Dictionary (WinUI 3)</value>
+    <value>Resource Dictionary</value>
   </data>
   <data name="1010" xml:space="preserve">
-    <value>[Experimental] Resource Dictionary (WinUI 3)</value>
+    <value>[Experimental] Resource Dictionary</value>
   </data>
   <data name="1011" xml:space="preserve">
-    <value>An empty, keyed collection of XAML resources for apps based on the Windows UI Library (WinUI 3).</value>
+    <value>An empty, keyed collection of XAML resources (WinUI 3).</value>
```
Comment (Lamparter) [inline dev/VSIX/Extension/Cpp/Common/VSPackage.resx:145]:
```suggestion
    <value>An empty, keyed collection of XAML resources (WinUI).</value>
```

File: dev/VSIX/Extension/Cpp/Common/VSPackage.resx [hunk 9]
```diff
@@ -151,58 +151,58 @@
     <value>[Experimental] Resources File (.resw)</value>
   </data>
   <data name="1015" xml:space="preserve">
-    <value>A file for storing string and conditional resources for apps based on the Windows UI Library (WinUI 3).</value>
+    <value>A file for storing string and conditional resources (WinUI 3).</value>
```
Comment (Lamparter) [inline dev/VSIX/Extension/Cpp/Common/VSPackage.resx:154]:
```suggestion
    <value>A file for storing string and conditional resources (WinUI).</value>
```

File: dev/VSIX/Extension/Cpp/Common/VSPackage.resx [hunk 10]
```diff
@@ -151,58 +151,58 @@
...
-    <value>User Control (WinUI 3)</value>
+    <value>User Control</value>
   </data>
   <data name="1022" xml:space="preserve">
-    <value>[Experimental] User Control (WinUI 3)</value>
+    <value>[Experimental] User Control</value>
   </data>
   <data name="1023" xml:space="preserve">
-    <value>A user control for apps based on the Windows UI Library (WinUI 3).</value>
+    <value>A blank user control (WinUI 3).</value>
```
Comment (Lamparter) [inline dev/VSIX/Extension/Cpp/Common/VSPackage.resx:172]:
```suggestion
    <value>A blank user control (WinUI).</value>
```

**Review by Lamparter — COMMENTED** [review]
Maybe just "WinUI" is better?

File: dev/VSIX/Extension/Cs/Common/VSPackage.resx (hunk 1 above)
Comment (0x5bfa) [reply dev/VSIX/Extension/Cs/Common/VSPackage.resx:184]: Yeah

File: dev/VSIX/Extension/Cpp/Common/VSPackage.resx [hunk 11]
```diff
@@ -151,58 +151,58 @@
...
-    <value>Blank App, Packaged (WinUI 3 in Desktop)</value>
+    <value>WinUI Blank App (Packaged)</value>
   </data>
   <data name="1030" xml:space="preserve">
-    <value>[Experimental] Blank App, Packaged (WinUI 3 in Desktop)</value>
+    <value>[Experimental] WinUI Blank App (Packaged)</value>
   </data>
   <data name="1031" xml:space="preserve">
-    <value>A project template for creating a Desktop app based on the Windows UI Library (WinUI 3) along with a MSIX package for side-loading or distribution via the Microsoft Store.</value>
+      <value>A project template for creating a WinUI 3 app along with a MSIX package for side-loading or distribution via the Microsoft Store.</value>
```
Comment (Lamparter) [inline dev/VSIX/Extension/Cpp/Common/VSPackage.resx:190]:
```suggestion
    <value>A project template for creating a WinUI 3 app along with a MSIX package for side-loading or distribution via the Microsoft Store.</value>
```

**Comment by niels9001** [comment]
Closing this PR in favor of: #4986

File: dev/VSIX/ProjectTemplates/Desktop/CSharp/PackagedApp/BlankApp/WinUI.Desktop.Cs.BlankApp.vstemplate [hunk 12]
```diff
@@ -2,8 +2,8 @@
 <VSTemplate Version="3.0.0" xmlns="http://schemas.microsoft.com/developer/vstemplate/2005" Type="Project">
   <TemplateData>
     <Hidden>true</Hidden>
-    <Name>Blank App (WinUI 3 in Desktop)</Name>
-    <Description>A project for creating a Desktop app based on the Windows UI Library (WinUI 3).</Description>
+    <Name>WinUI Blank App</Name>
```
Comment (npuvvada) [inline dev/VSIX/ProjectTemplates/Desktop/CSharp/PackagedApp/BlankApp/WinUI.Desktop.Cs.BlankApp.vstemplate:5]: Can we use resx file for Name, description in these templates as well ?

File: dev/VSIX/Extension/Cpp/Common/VSPackage.resx [hunk 13]
```diff
@@ -151,58 +151,58 @@
...
-    <value>[Experimental] Blank App, Packaged (WinUI 3 in Desktop)</value>
+    <value>[Experimental] WinUI Blank App (Packaged)</value>
   </data>
   <data name="1031" xml:space="preserve">
-    <value>A project template for creating a Desktop app based on the Windows UI Library (WinUI 3) along with a MSIX package for side-loading or distribution via the Microsoft Store.</value>
+    <value>A project template for creating a WinUI 3 app along with a MSIX package for side-loading or distribution via the Microsoft Store.</value>
   </data>
   <data name="1033" xml:space="preserve">
-    <value>Windows Runtime Component (WinUI 3)</value>
+    <value>Windows Runtime Component</value>
```
Comment (npuvvada) [inline dev/VSIX/Extension/Cpp/Common/VSPackage.resx:193]: Any specific reason for removing WinUI3 as I see there are windows runtime components for UWP as well and keeping WinUI 3 might help to distinguish.

**Pull request closed without merging**